src/main/core/Event/CatalogEvents/FunctionalEvents.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Claroline\CoreBundle\Event\CatalogEvents;
+
+final class FunctionalEvents
+{
+    /**
+     * @Event("Claroline\CoreBundle\Event\Functional\AddBadgeEvent")
+     */
+    public const ADD_BADGE = 'event.funcitonal.add_badge';
+
+    /**
+     * @Event("Claroline\CoreBundle\Event\Functional\RemoveBadgeEvent")
+     */
+    public const REMOVE_BADGE = 'event.funcitonal.remove_badge';
+
+    /**
+     * @Event("Claroline\CoreBundle\Event\Functional\ResourceOpenEvent")
+     */
+    public const RESOURCE_OPEN = 'event.functional.resource_open';
+
+    /**
+     * @Event("Claroline\CoreBundle\Event\Functional\ResourceEvaluationEvent")
+     */
+    public const RESOURCE_EVALUATION = 'event.functional.resource_evaluation';
+
+    /**
+     * @Event("Claroline\CoreBundle\Event\Functional\ResourceScoreEvent")
+     */
+    public const RESOURCE_SCORE = 'event.functional.resource_score';
+
+    /**
+     * @Event("Claroline\CoreBundle\Event\Functional\ToolOpenEvent")
+     */
+    public const TOOL_OPEN = 'event.functional.tool_open';
+}
```

Comment:
This should be broken into multiple files. There are too many events in the core to put them all inside one file.
Also `FunctionalEvents` doesn't give much information about what goes in. It's better too keep the functional structure : `ResourceEvents` / `ToolEvents` / `WorkspaceEvents`.

Reply:
According to the comment about not dispatching your own events, event names must be replaced by the existing ones.

Reply:
Done, thx :)